pulp_python/app/models.py

```diff
@@ -1,35 +1,73 @@
-from gettext import gettext as _
 from logging import getLogger
 
 from django.db import models
 
-from pulpcore.plugin.models import (Artifact, Content, ContentArtifact, RemoteArtifact, Importer,
-                                    ProgressBar, Publisher, RepositoryContent, PublishedArtifact,
-                                    PublishedMetadata)
-from pulpcore.plugin.tasking import Task
+from pulpcore.plugin.models import Content, Importer, Publisher
 
 
 log = getLogger(__name__)
 
 
-class PythonContent(Content):
+class Classifier(models.Model):
     """
-    The "python" content type.
+    Custom tags for classifier
 
-    Define fields you need for your new content type and
-    specify uniqueness constraint to identify unit of this type.
+    Fields:
 
-    For example::
+        name (models.TextField): The name of the classifier
 
-        field1 = models.TextField()
-        field2 = models.IntegerField()
-        field3 = models.CharField()
+    Relations:
 
-        class Meta:
-            unique_together = (field1, field2)
+        pythonpackagecontent (models.ForeignKey): The task this tag is associated with
+
+    """
+
+    name = models.TextField()
+    pythonpackagecontent = models.ForeignKey("PythonPackageContent", related_name="classifiers",
+                                             related_query_name="classifier")
+
+
+class PythonPackageContent(Content):
+    """
+    A Content Type representing Python's Distribution Package as
+    defined in pep-0426
     """
 
     TYPE = 'python'
+    filename = models.TextField(unique=True, db_index=True, blank=False)
+    path = models.TextField(blank=False)
+    packagetype = models.TextField(blank=False)
+    name = models.TextField(blank=False)
+    version = models.TextField(blank=False)
+
+    # Note that this field is not contained in PyPI nor Warehouse
+    # ignore this on publish to PyPI or Warehouse
+    # This is required per PEP426
+    metadata_version = models.TextField(blank=False)
+
+    summary = models.TextField()
+    description = models.TextField()
+    keywords = models.TextField()
+    home_page = models.URLField()
+    download_url = models.URLField()
+    author = models.TextField()
+    author_email = models.EmailField()
+    maintainer = models.TextField()
+    maintainer_email = models.EmailField()
+    license = models.TextField()
+    requires_python = models.TextField()
+    project_url = models.URLField()
+    platform = models.TextField()
+
+    # Note supported_platform exists in PEP0345 but not in legacy-PyPI or warehouse
+    # ignore this on publish to PyPI or Warehouse
+    supported_platform = models.TextField()
+
+    # These should be reference to other packages, treat them as text fields for now
```

Comment:
I don't think these can reference other packages really. For instance, scipy returns "numpy (>=1.8.2)" for "requires_dist". I think text is all we can do now, and probably all we can do for now.

Reply:
I think we should be able to do dependency solving. It's my hope that we can specify "sync this and all dependencies" to pulp. An have this be links to the various dependencies.

Reply:
The trouble is that ">" can't be a link. These are closer to queries than links.